src/main/java/org/venice/beachfront/bfapi/config/BfApiConfig.java

```diff
@@ -0,0 +1,76 @@
+package org.venice.beachfront.bfapi.config;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.Authentication;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.Cloud;
+import org.springframework.cloud.CloudFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.joda.JodaModule;
+
+import io.swagger.annotations.ApiOperation;
+import springfox.documentation.builders.ApiInfoBuilder;
+import springfox.documentation.builders.PathSelectors;
+import springfox.documentation.builders.RequestHandlerSelectors;
+import springfox.documentation.service.ApiInfo;
+import springfox.documentation.service.Contact;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+
+@Configuration
+public class BfApiConfig {
+
+	@Bean
+	public ObjectMapper getJacksonObjectMapper() {
+		ObjectMapper mapper = new ObjectMapper();
+		mapper.registerModule(new JodaModule());
+		return mapper;
+	}
+
+	@Bean
+	public ExecutorService getExecutor(@Value("${concurrent.threads}") int threads) {
+		return Executors.newFixedThreadPool(threads);
+	}
+
+	@Bean
+	public Docket beachfrontApi() {
+		// @formatter:off
+		final ApiInfo apiInfo =
+				new ApiInfoBuilder()
+					.title("Beachfront API")
+					.description("Beachfront Web Services")
+					.contact(new Contact("The VeniceGeo Project", "http://radiantblue.com", "venice@radiantblue.com"))
+					.version("0.1.0")
+					.build();
+
+		return new Docket(DocumentationType.SWAGGER_2)
+				.useDefaultResponseMessages(false)
+				.ignoredParameterTypes(Authentication.class)
+				.groupName("Beachfront")
+				.apiInfo(apiInfo)
+				.select()
+				.apis(RequestHandlerSelectors.withMethodAnnotation(ApiOperation.class))
+				.paths(PathSelectors.any())
+				.build();
+		// @formatter:on
+	}
+
+	@Bean
+	public Cloud cloud() {
+		return new CloudFactory().getCloud();
+	}
+
+	@Bean
+	@ConfigurationProperties
+	public DataSource dataSource() {
+		return cloud().getServiceConnector("pz-postgres", DataSource.class, null);
```

Comment:
This line is new - can you explain a bit, in a comment, about what it does, and why it's needed?

Reply:
Without specifying which PCF service to bind for the DataSource, we see the error below on deployment. The `spring-cloud-sso-connector` library interferes here somehow.

`2018-05-21T16:51:23.652-04:00 [APP/PROC/WEB/0] [OUT] 2018-05-21 20:51:23.652 WARN 21 --- [ main] ationConfigEmbeddedWebApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.NoUniqueBeanDefinitionException: No qualifying bean of type 'javax.sql.DataSource' available: expected single matching bean but found 2: pz-postgres,pz-postgres-service-key`

Reply:
How will this affect local development? When we develop locally, no one is going to have a running pz-postgres or pz-postgres-service-key VCAP environment.

Reply:
Setting up fake local VCAP variables shouldn't be that bad. I could whip up a Bash shell script that does that, if we need it.

Reply:
Is there a way we can achieve this WITHOUT having to have a local vcap tree set up? I really, really, really would like to avoid that if possible.

We don't actually need the pz-postgres-service-key Data connector set up. This is used purely in Liquibase and GeoServer via config file reads. Instead of enabling the pz-postgres (and thus developers are required to populate it) then can we explicitly DISABLE the pz-postgres-service-key? That'll maintain functionality without us having to set up a VCAP tree.

Reply:
I will limit this Bean configuration to the `cloud` profile. Will that suffice?